exercises/practice/satellite/lib/satellite.ex

```diff
@@ -0,0 +1,14 @@
+defmodule Satellite do
+  @typedoc """
+  A tree, which can be empty, or made from a left branch, a node and a right branch
+  """
+  @type tree :: {} | {tree, any, tree}
+
+  @doc """
+  Build a tree from the elements given in a pre-order and in-order style
+  """
+  @spec build_tree(preorder :: [any], inorder :: [any]) :: tree | {:error, String.t()}
```

Comment:
I forgot to pay attention to this in the previous PRs: the more "Elixir way" would be to return:
```
{:ok, tree} | {:error, String.t()}
```
Usually functions that return an error tuple when something goes wrong, return an ok-tuple when everything is fine (a few examples: [File.read](https://hexdocs.pm/elixir/File.html#read/1), [DateTime.new](https://hexdocs.pm/elixir/DateTime.html#new/2), [Ecto.Repo.insert](https://hexdocs.pm/ecto/Ecto.Repo.html#c:insert/2)). It would be nice if our exercises followed that convention too.

It would be great if you did that change in POV too. Better in a separate PR, that will give you extra reputation points 😬 (reputation is a new feature in v3, it's just some points that you can collect but not really use for anything).

Reply:
OK. Actually at some point I had an implementation that used `{:ok, tree}` and all the checks along the way used `with` instead of using `MapSet` but I found it a bit bulky. It was the better design in the end.

I'll revisit POV too, what wouldn't I do for useless internet points :)